src/java.compiler/share/classes/javax/lang/model/util/Types.java

```diff
@@ -226,6 +234,8 @@ public interface Types {
      * {@return a new wildcard type}  Either of the wildcard's
      * bounds may be specified, or neither, but not both.
      *
+     * Annotations on the bounds are preserved.
+     *
      * @param extendsBound  the extends (upper) bound, or {@code null} if none
      * @param superBound    the super (lower) bound, or {@code null} if none
      * @throws IllegalArgumentException if bounds are not valid
```

Comment:
Should annotation also be preserved (and also documented to be preserved) for `getDeclaredType`?

Reply:
@jddarcy ?

Reply:
I don't think it is preserved in that case as much as a lookup/construction process is done based on the TypeElement, etc.

So I think leaving the annotation behavior unspecified is okay.

Reply:
The `getDeclaredType` accepts `TypeMirror`s as the type arguments. I tried to invoke the method when this `TypeMirror` has an annotation, and the annotation is in the resulting `DeclaredType` as well. I don't mind this behavior, but my reading of the class' javadoc is that there should be no annotations, unless otherwise stated. And there does not seem to be anything in the `getDeclaredType` javadoc related to annotations on the type arguments, so the class' javadoc would apply. But maybe I miss something/read something wrong?

Reply:
Ah, right; the type arguments. I was thinking of the root return type. Specification and tests updated. Thanks.